### ISSUE TITLE
Joystick lightgun option

### ISSUE THREAD

**Issue by Somechump** [libretro/RetroArch#13425]
The option for controlling a lightgun gun game on retroarch is limited to mouse or lightgun. 

Please add another input option that gives the ability to map a joystick axis to also be used as a lightgun input. This will need be be mapped as the whole axis e.g. x axis, y axis and not how retroarch currently maps input with x-left, x-right, y-up, y-down. 

This will allow multiplayer lightgun games instead of having a single mouse input. 

This is currently possible in stand alone emulators such as Mednafen, Supermodel, Teknoparrot, Mame (including retroarch Mame, but configured inside the core) and possibly others. These emulators allow the binding of a joystick axis to allow the lightgun cursor to be positioned anywhere on screen by holding a joystick in any position. Demulshooter also retroactively adds this feature into other emulators that do not support this feature. 
These emulators all work great with a wiimote and dolphinbar which makes the ir pointer the z and zr axis. 

For your consideration.

**Comment by MrHuu** [comment]
The title may be confusing. This is about being able to control 'lightgun games' with a controller. Currently the 'lightgun' support only takes input from mouse or touchscreen coordinates.

The request is about adding the controller thumbstick axis as a form of input to control lightgun games. Not the other way around.
Which is a pretty decent request in my opinion.

Unfortunately, most cores have hardcoded lightgun support only taking input from a mouse. Instead of using the libretro 'lightgun' implementation which could be redirected to any platform supported form of input such as a controller.

With the lightgun trigger only taking input from the ```left_mouse_button``` instead of ```lightgun_trigger``` for example.
Not being able to map a controller button to ```lightgun_trigger```.

Currently most cores decide which input to take, ignoring the frontend input driver.
While the frontend lightgun support is available, cores haven't been updated to reflect this support.


For example, PCSX-ReARMed:
https://github.com/libretro/pcsx_rearmed/blob/master/frontend/libretro.c#L2247-L2287

**Comment by StormedBubbles** [comment]
Hi,

I am not sure about getting thumbsticks to control the Libretro lightgun coordinates specifically, but almost all console games should have a controller option for lightgun games already. Atari 2600 (lr-stella) is the only one I can think of that has no such option, but that list of games is 1 officially released + 1 unreleased + a handful of homebrew games. The various MAME versions have options for controlling the gun axes with controllers. The analog-stick support may not work very well with the true lightgun games, but you can usually map the stick to digital input and adjust sensitivities to make those games play like console lightgun games with a controller.

Is there a particular benefit for using the thumbstick to specifically control the lightgun input?

P.S. I recently added support for mappable lightgun buttons to lr-pcsx-rearmed. If you find any other emulators that hard code the lightgun buttons, I can adjust those to make the buttons mappable.

**Comment by DamianS-eng** [comment]
This is not a viable request for multiplayer lightgun titles, only useful if the system cannot have a mouse or mouse-like input in any way.

Even if the feature was implemented, it would be severely restricted by the range limitations of joystick axes and the fact that joysticks cannot report true accurate displacement or absolute coordinates, two qualities required for mouse or lightgun software.

Rather, pursue methods to allow multiple mice, not tied to the OS' default mouse. Having an option to have the second player use their cursor controlled by the joystick severely hinders their experience compared to the mouse user.

**Comment by lazzer2000** [comment]
it could with support for the drivers that have been written for the Guncon3 also. it seems to appear as a joystick instead of mouse/Pointing device.

**Comment by Trihy** [comment]
This could solve many problems. With mame core you can set axis Y X to control lightgun. So two player with xinput analog stick is easy and works pretty good with wiimotes.

Not sure how can this be added to platforms which use the analog stick for other purposes. 

But for platforms that don't use analog stick (new, snes, genesis) this could solve the multiplayer problems and make it easier to config.

**Comment by rcl0ng** [comment]
I agree with Trihy. I've had very good luck using left stick as gun with many emulators.  I use wiimotes and the lightgun software 'lichtknarre'.  This setup works with many from flycast to mednafen.  The only core that has this feature in retroarch is fbneo nes core.  it would be nice to see it added to other cores.
